jupyter_server/services/shutdown.py

```diff
@@ -8,8 +8,12 @@
 
 class ShutdownHandler(JupyterHandler):
     @web.authenticated
-    def post(self):
+    async def post(self):
         self.log.info("Shutting down on /api/shutdown request.")
+
+        await self.terminal_manager.terminate_all()
+        await self.kernel_manager.shutdown_all()
+
```

Comment:
```suggestion

        await self._cleanup()

```

Reply:
Actually this doesn't work. The handler does not inherit from the ServerApp, we don't have a cleanup method.